src/includes/form.inc.js

```diff
@@ -24,7 +24,10 @@ function _drupalgap_form_add_another_item(form_id, name, delta) {
     form.elements[name][language][delta + 1] = item;
     var element = form.elements[name];
     var variables = {
-      attributes: {},
+      attributes: {
+        id: item.id,
```

Comment:
the delta value is appended to the id by `drupalgap_form_element_item_create()` at line 19.